hyperspy/signal.py

```diff
@@ -2248,6 +2248,34 @@ def _data_aligned_with_axes(self):
             data = self.data.transpose(nav_iia_r + sig_iia_r)
             return data
 
+    def _rebin_validate_and_get_factors(self, new_shape=None, scale=None,
+                                        crop=True, out=None):
+
+        if new_shape is None and scale is None:
+            raise ValueError("One of new_shape, or scale must be specified")
+        elif new_shape is None and scale is None:
+            raise ValueError("Only one out of new_shape or scale should be specified.\
+                            Not both.")
+        elif new_shape:
+            if len(new_shape) != len(self.data.shape):
+                raise ValueError("Wrong new_shape size")
+            new_shape_in_array = []
+            for axis in self.axes_manager._axes:
+                new_shape_in_array.append(
+                    new_shape[axis.index_in_axes_manager])
+            factors = (
+                np.array(
+                    self.data.shape) /
+                np.array(new_shape_in_array))
+        else:
+            if len(scale) != len(self.data.shape):
+                raise ValueError("Wrong scale size")
+            factors = []
+            for axis in self.axes_manager._axes:
+                factors.append(scale[axis.index_in_axes_manager])
+            factors = np.array(factors)
+        return factors  # Factors are in array order
```

Comment:
I would still prefer this to be outside the class. Among other reasons, it currently is a method with a name that reads something like "rebin and validate and get factors" - which is confusing, even though I know what it does.

Also, currently `crop=True, out=None` are not used, and a quick docstring would be useful

I think ideally it would be something like
```python
factors = _function_name(new_shape, scale, data_shape, axes_order)
```
where 
```python
axes_order = [axis.index_in_axes_manager for axis in self.axes_manager._axes]
```